src/main/kotlin/net/minecrell/pluginyml/paper/PaperPluginDescription.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Minecrell <https://github.com/Minecrell>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package net.minecrell.pluginyml.paper
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.annotation.JsonSerialize
+import groovy.lang.Closure
+import net.minecrell.pluginyml.PluginDescription
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Nested
+import org.gradle.api.tasks.Optional
+
+class PaperPluginDescription(project: Project) : PluginDescription {
+
+    @Input @JsonProperty("api-version") var apiVersion: String? = null
+    @Input var name: String? = null
+    @Input var version: String? = null
+    @Input var main: String? = null
+    @Input @JsonIgnore var generatePluginLibraries: Boolean = false
+    @Input @Optional var bootstrapper: String? = null
+    @Input @Optional var loader: String? = null
+    @Input @Optional var description: String? = null
+    @Input @Optional var load: PluginLoadOrder? = null
+    @Input @Optional var author: String? = null
+    @Input @Optional var authors: List<String>? = null
+    @Input @Optional var contributors: List<String>? = null
+    @Input @Optional var website: String? = null
+    @Input @Optional var prefix: String? = null
+    @Input @Optional @JsonProperty("default-permission") var defaultPermission: Permission.Default? = null
```

Comment:
Not sure why but this seems to be called

```suggestion
    @Input @Optional @JsonProperty("defaultPerm") var defaultPermission: Permission.Default? = null
```

https://github.com/PaperMC/Paper/blob/29b17a892d11697dce9ee35d8eab593bb4748fb1/patches/server/0013-Paper-Plugins.patch#L4800
https://github.com/PaperMC/Paper/blob/29b17a892d11697dce9ee35d8eab593bb4748fb1/test-plugin/src/main/resources/paper-plugin.yml#L10